lib/sync/CoreIdExtension.js

```diff
@@ -0,0 +1,39 @@
+/**
+* @module CoreIdExtension
+*/
+export class CoreIdExtension {
+  /** @type {Hypercore} */
+  #core
+  /** @type {import('./CoreIdCache.js').CoreIdCache} */
+  #coreIdCache
+  /** @type {any} */
+  #extension
+
+  /**
+  * @callback OnMessage 
+  * @param {CoreId[]} coreIds - list of coreIds from 'auth'
+  * @param {Object} peer - socket peer
+  */
+  /**
+  * Create a CoreReplicator instance. This class is in charge of sharing a set of cores by registering an hypercore extension.
+  * @param {Hypercore} core - core to use as master core to register the extension in
+  * @param {import('./CoreIdCache.js').CoreIdCache} coreIdCache the cache to retrieve the set of coreIds to share
+  * @param {OnMessage} [onmessage] - callback that handles recieved messages from a peer
+  */
+  constructor(core, coreIdCache, onmessage = () => {}){
+    this.#core = core 
```

Comment:
It's really important that these extensions are _only_ added to the project creator auth store core (the core that is used for the initial connection). We should document that somewhere clearly.